README.md

```diff
@@ -411,40 +413,39 @@ numberInput = do
 Let's give it some html attributes to work with:
 
 ```haskell
-numberInput :: MonadWidget t m => m (Dynamic t (Maybe Double))
+numberInput :: DomBuilder t m => m (Dynamic t (Maybe Double))
 numberInput = do
-  let initAttrs = (("type" =: "number") <> ("style" =: "border-color: blue"))
+  let initAttrs = ("type" =: "number") <> ("style" =: "border-color: blue")
   n <- inputElement $ def
     & inputElementConfig_initialValue .~ "0"
     & inputElementConfig_elementConfig . elementConfig_initialAttributes .~ initAttrs
   return . fmap (readMaybe . unpack) $ _inputElement_value n
 ```
 
-Here, we've used a `(Map Text Text)`. This `Map` represents the html attributes of our inputs. 
+Here, we've used a `Map Text Text`. This `Map` represents the html attributes of our inputs.
 
 Static attributes are useful and quite common, but attributes will often need to change.
 Instead of just making the `InputElement` blue, let's change it's color based on whether the input successfully parses to a `Double`:
 
 ```haskell
 {-# LANGUAGE RecursiveDo #-}
+import Control.Monad.Fix (MonadFix)
 ...
-numberInput :: (MonadWidget t m) => m (Dynamic t (Maybe Double))
+numberInput :: (DomBuilder t m, MonadFix m) => m (Dynamic t (Maybe Double))
 numberInput = do
   let initAttrs = ("type" =: "number") <> (style False)
       color error = if error then "red" else "green"
       style error = "style" =: ("border-color: " <> color error)
       styleChange :: Maybe Double -> Map AttributeName (Maybe Text)
       styleChange result = case result of
-        (Just _) -> fmap Just (style False)
-        (Nothing) -> fmap Just (style True)
-
-  rec
```

Comment:
I was getting ghcjs parse errors with rec like this on separate line. I'm not yet experienced with the RecursiveDo syntax, but the way I changed it in the PR worked fine.